lib/resque.rb

```diff
@@ -255,6 +295,30 @@ def after_pause=(block)
     register_hook(:after_pause, block)
   end
 
+  def before_reserve(&block)
+    block ? register_hook(:before_reserve, block) : hooks(:before_reserve)
+  end
+
+  def before_reserve=(block)
+    register_hook(:before_reserve, block)
+  end
+
+  def after_perform(&block)
+    block ? register_hook(:after_perform, block) : hooks(:after_perform)
+  end
+
+  def after_perform=(block)
+    register_hook(:after_perform, block)
+  end
```

Comment:
FEATURE: global `after_perform`

The monolith uses this callback to decide when to shut down workers that consume too much RAM.

Could be PRed to master.